paasta_tools/spark_tools.py

```diff
@@ -264,7 +264,21 @@ def get_spark_memory_in_unit(mem: str, unit: Literal["k", "m", "g", "t"]) -> flo
         try:
             memory_bytes = float(mem[:-1]) * MEM_MULTIPLIER[mem[-1]]
         except (ValueError, IndexError):
-            print(f"Unable to parse memory value {mem}. Defaulting to 2 GB.")
-            memory_bytes = 2147483648  # default to 2 GB
+            print(f"Unable to parse memory value {mem}. Defaulting to 0.")
+            memory_bytes = 0  # default to 0
     memory_unit = memory_bytes / MEM_MULTIPLIER[unit]
     return memory_unit
+
+
+def get_spark_driver_memory_overhead_in_mb(spark_config: Dict[str, str]) -> float:
+    """
+    Returns the Spark driver memory overhead in bytes.
+    """
+    driver_mem_overhead = "0"
```

Comment:
We'll need to also consider `spark.driver.memoryOverheadFactor`, it can be set by the user or from the default in Spark (0.1)

Reply:
We'll consider that in service_configuration_lib, which will spit out a final `spark.driver.memoryOverhead`. I don't want to add all that logic here

Reply:
Do you mean overwriting the `spark.driver.memoryOverhead` value after considering `spark.driver.memoryOverheadFactor` in service_configuration_lib? This doesn't sound better than reading both of them together to me